source/agora/node/Validator.d

```diff
@@ -86,6 +94,58 @@ public class Validator : FullNode, API
         // currently we are not saving preimage info,
         // we only have the commitment in the genesis block
         this.regenerateQuorums(Height(0));
+
+        if(config.node.registry_address.length && config.addresses_to_register.length)
```

Comment:
Any usage of vibe.d should be wrapped so we can use LocalRest during unittest runs and only use vibe.d in non-unittest builds. That's one of the items in #1173:

> It needs to integrate with the existing testsuite (localrest, integration);

There should likely be something like `getClient`, perhaps called `getNameRegistryClient` in the `NetworkManager` which by defaults instantiates a `RestInterfaceClient!NameRegistryAPI`, but is overriden in unittests in `Base.d` to provide a non-network name registry node - similar to how it's done [here](https://github.com/bpfkorea/agora/blob/0a45c1359a9c819acae588aae25036f00c33e4af/source/agora/test/Base.d#L858). That code provides a LocalRest-based interface to the node, while the node itself is spawned for example [here](https://github.com/bpfkorea/agora/blob/0a45c1359a9c819acae588aae25036f00c33e4af/source/agora/test/Base.d#L638). For the name server node we could spawn it just once, for example somewhere inside of [here](https://github.com/bpfkorea/agora/blob/0a45c1359a9c819acae588aae25036f00c33e4af/source/agora/test/Base.d#L1338) and change the node's `registry_address` to match the same string as set with a call to `reg.register` like [here](https://github.com/bpfkorea/agora/blob/0a45c1359a9c819acae588aae25036f00c33e4af/source/agora/test/Base.d#L654).

A bit of pseudo-code:

```D
Base.d:

class TestAPIManager
{
    void makeNameRegistry ()
    {
        // TestNameRegistry would be the actual class which implements NameRegistryAPI,
        // just using a hashmap internally
        auto tid = RemoteAPI!NameRegistryAPI.spawn!TestNameRegistry(ctor_args...);
        this.reg.register("name.registry", api.tid());
    }
}

class TestNetworkManager
{
    /// get a localrest-backed client
    TestAPI getNameRegistryClient (Address address,
        Duration timeout)
    {
        auto tid = this.registry.locate(address);
        if (tid != typeof(tid).init)
            return new RemoteAPI!getNameRegistryAPI(tid, timeout);
        assert(0, "Trying to access node at address '" ~ address ~
               "' without first creating it");
    }
}

function makeTestNetwork ()
{
    // for each node's config set its `registry_address` to "name.registry" as set above
    foreach (node; nodes..)
        ...
    
    network.makeNameRegistry();
}
```

Reply:
> ```d
> makeTestNetwork
> ```

thanks, I will modify the code according to this